pybamm/solvers/idaklu_solver.py

```diff
@@ -675,3 +676,41 @@ def _integrate(self, model, t_eval, inputs_dict=None):
             return newsol
         else:
             raise pybamm.SolverError("idaklu solver failed")
+
+    def jaxify(
+        self,
+        model,
+        t_eval,
+        *,
+        output_variables=None,
+        inputs=None,
+        calculate_sensitivities=True,
+    ):
+        """JAXify the solver object
+
+        Creates a JAX expression representing the IDAKLU-wrapped solver
+        object.
+
+        Parameters
+        ----------
+        model : :class:`pybamm.BaseModel`
+            The model to be solved
+        t_eval : numeric type, optional
+            The times at which to compute the solution. If None, the times in the model
+            are used.
+        output_variables : list of str, optional
+            The variables to be returned. If None, the variables in the model are used.
+        inputs : dict, optional
+            Any inputs to the model
+        calculate_sensitivities : bool, optional
```

Comment:
Yes, I wasn't sure whether to retain this argument or not, since it's value can be trivially determined based on whether `inputs` are present during the solve. However, it does form part of the API for the IDAKLU solver, and it is functional (you can specify `False` to prevent sensitivities from being calculated, in which case `grad` will fail with a `ValueError: Cannot compute sensitivities. The 'calculate_sensitivities' argument of the solver.solve should be changed from 'None' to allow sensitivities calculations. Check solver documentation for details.`; by default the argument is `True`). I would be just as happy to remove it as an argument from the interface though.

Reply:
happy to keep it, all sounds good